### PR TITLE
Support python3.11 and bump version

### PR DESCRIPTION
**Pull request by geek-yang**
- Drop python 3.8 (since the latest version of `xarray` not support 3.8 anymore)
- Support python 3.11
- Bump version to prepare for a new release

### CODE REVIEW

**Comment by geek-yang** [comment]
Alright...the failing CI seems to imply that the wheel of `numpy` for python 3.11 is not available yet. Found some similar issue with py3.10 before (check [this](https://stackoverflow.com/questions/69458399/numpy-1-21-2-may-not-yet-support-python-3-10))...Update `pip` and pin version of `numpy` do not help. But this is very strange...As in `zampy` it works and we have the same setup for CD/CI and project.toml. Any idea about it? @BSchilperoort

File: pyproject.toml [hunk 1]
```diff
@@ -22,7 +22,7 @@ name = "lilio"
 description = "python package for generating calendars for machine learning timeseries analysis."
 readme = "README.md"
 license = "Apache-2.0"
-requires-python = ">=3.8, <3.11"
+requires-python = ">=3.9, <=3.11"
```
Comment (BSchilperoort) [inline pyproject.toml:25]:
```suggestion
requires-python = ">=3.9, <3.12"
```
This should fix the problem.

Comment (BSchilperoort) [reply pyproject.toml:25]: The hatch runners didn't give a clear error, but a local 3.11 environment test gave the following error:
```
ERROR: Package 'lilio' requires a different Python: 3.11.5 not in '<=3.11,>=3.9'
```

Comment (geek-yang) [reply pyproject.toml]: 😂 Thanks for the solution @BSchilperoort ! I never realized that `python3.11` would exclude `python3.11.x` 😮!

File: CHANGELOG.md [hunk 2]
```diff
@@ -6,6 +6,16 @@ and this project adheres to [Semantic Versioning](https://semver.org/).
 
 ## [Unreleased]
 
+## 0.4.1 (2023-09-11)
+### Added
+- Python 3.11 support ([#60](https://github.com/AI4S2S/lilio/pull/60)).
+
+### Changed
+- Improved dask ingegration guide ([#59](https://github.com/AI4S2S/lilio/pull/59)).
```
Comment (BSchilperoort) [inline CHANGELOG.md:14]:
```suggestion
- Improved dask integration guide ([#59](https://github.com/AI4S2S/lilio/pull/59)).
```

**Review by BSchilperoort — APPROVED** [review]
Just spotted a typo, otherwise it looks good.

Perhaps it can also be good to set mypy's python version? I believe it spots old syntax that can now be typed in a nicer way (such as `List` -> `list`, `Union[a, b]` to `a | b`).

**Comment by geek-yang** [comment]
> Just spotted a typo, otherwise it looks good.
> 
> Perhaps it can also be good to set mypy's python version? I believe it spots old syntax that can now be typed in a nicer way (such as `List` -> `list`, `Union[a, b]` to `a | b`).

I thought about it indeed, but the type union `|` thingy is supported from python 3.10 (https://docs.python.org/3/whatsnew/3.10.html), and it does not work with python 3.9, which could cause issue for CD/CI, right?

File: CHANGELOG.md (hunk 2 above)
Comment (geek-yang) [reply CHANGELOG.md:14]: Well spot! Thanks @BSchilperoort 😂

**Comment by BSchilperoort** [comment]
> I thought about it indeed, but the type union `|` thingy is supported from python 3.10 (https://docs.python.org/3/whatsnew/3.10.html), and it does not work with python 3.9, which could cause issue for CD/CI, right?

I just meant it as an example, so if we change the mypy version (eventually) to 3.10, then mypy will tell us we can use the `|` as OR.

**Comment by geek-yang** [comment]
> > I thought about it indeed, but the type union `|` thingy is supported from python 3.10 (https://docs.python.org/3/whatsnew/3.10.html), and it does not work with python 3.9, which could cause issue for CD/CI, right?
> 
> I just meant it as an example, so if we change the mypy version (eventually) to 3.10, then mypy will tell us we can use the `|` as OR.

I got it now! That sounds good! I will add a line `python_version = "3.10"`

**Pull request merged**